source/write-operations/replace-documents.txt

```diff
@@ -0,0 +1,212 @@
+.. _java-rs-write-replace:
+
+=================
+Replace Documents
+=================
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 2
+   :class: singlecol
+
+.. facet::
+   :name: genre
+   :values: reference
+
+.. meta::
+   :keywords: code examples, write, replace, change
+
+Overview
+--------
+
+In this guide, you can learn how to use the {+driver-short+} to replace
+a document in a MongoDB collection by performing replace operations.
+
+A replace operation replaces one document in a MongoDB collection.
+You can perform a replace operation by using the ``replaceOne()`` method.
+
+Sample Data
+~~~~~~~~~~~
+
+The examples in this guide use the ``sample_restaurants.restaurants`` collection
+from the :atlas:`Atlas sample datasets </sample-data>`. 
+
+To learn how to create a free MongoDB Atlas cluster and load the sample
+datasets, see the :ref:`<java-rs-getting-started>` tutorial.
+
+.. include:: /includes/reactor-note.rst
+
+Replace Operation
+-----------------
+
+You can perform a replace operation in MongoDB by using the ``replaceOne()``
+method. This method removes all fields except the ``_id`` field from the first
+document that matches the search criteria. It then inserts the fields and values
+you specify into the document.
+
+Required Parameters
+~~~~~~~~~~~~~~~~~~~
+
+The ``replaceOne()`` method requires the following parameters:
+
+- A **query filter** document, which determines which documents to replace. For
+  more information about query filters, see the 
+  :manual:`Query Filter Documents section </core/document/#query-filter-documents>` in
+  the MongoDB Server manual. 
+
+- A **replace** document, which specifies the fields and values to insert in the new 
+  document.
+
+Replace One
+-----------
+
+To replace a single document in a MongoDB collection, call the ``replaceOne()`` method
+and pass the document you want to add. Then, pass the ``replaceOne()`` method to the static
+``Mono.from()`` method from the ``Mono Publisher``. The ``Mono Publisher`` is a
+``Publisher`` implementation from the Project Reactor library. In {+java-rs+}, you must use
```

Comment:
S: could we make a source constant for "Project Reactor library"? It could save some effort in spellchecking etc. 
maybe `prl`

Reply:
Or even just project reactor